### PR TITLE
Fix undeclared properties in hm-feed.php

### PR DESCRIPTION
**Pull request by josaphatim**
Fixes `PHP Fatal error:  Uncaught Error: Call to undefined method Hm_Contact::dump() in /path/to/cypht/lib/repository.php:59` and creation of dynamic proporties

### CODE REVIEW

File: lib/repository.php [hunk 1]
```diff
@@ -55,7 +55,12 @@ public static function add($entity, $save = true) {
 
     public static function edit($id, $entity) {
         if (array_key_exists($id, self::$entities)) {
-            self::$entities[$id] = array_merge(self::$entities[$id], $entity);
+            if (is_array(self::$entities[$id])) {
+                $data =  array_merge(self::$entities[$id], $entity);;
```
Comment (kroky) [inline lib/repository.php:59]: Please fix the double semicolon at the end.

File: lib/repository.php [hunk 2]
```diff
@@ -55,7 +55,12 @@ public static function add($entity, $save = true) {
...
+            } else {
+                $data = $entity;
```
Comment (kroky) [inline lib/repository.php:61]: When that's an object, don't we still need a merge? Otherwise, it is an overwrite.

Comment (josaphatim) [reply lib/repository.php:61]: when it is an object it already comes with the necessary modifications unlike an array where only certain properties can be modified

File: lib/repository.php (hunk 1 above)
Comment (josaphatim) [reply lib/repository.php]: fixed

File: lib/repository.php (hunk 2 above)
Comment (kroky) [reply lib/repository.php:61]: Ok but this makes the whole method quite confusing - imagine someone trying to use it - how dothey know if this method actually replaces the entity or updates the passed entity params only? I still suggest you unify - either do a full replace and require calling code to pass full entities or merge. Having a different business logic based on type of incoming parameters is a bad programming practice.

Comment (josaphatim) [reply lib/repository.php:61]: The same thing has been done here and merged

https://github.com/cypht-org/cypht/pull/1132/files#diff-e8ea654d76e94bfce8824e84e96fec6cf4d3497b6e3d446ab8e1f81d8eb983ceR58

**Comment by marclaporte** [comment]
@josaphatim This branch has conflicts that must be resolved

**Comment by josaphatim** [comment]
> @josaphatim This branch has conflicts that must be resolved

Fixed, corrected and ready to be merged

**Pull request merged**